# Project 6 -- Niharika Rai

**TA Help:** (for instance) John Smith, Alice Jones, etc., list names of any TAs who helped you

- For example: Help with figuring out how to write a function (describe the tasks that they helped you with)

**Collaboration:** My Friend in CS, My Uncle, Another Student, etc., list names of any other people who helped you

(describe the tasks that they helped you with)
- For example: helped figuring out how to load the dataset.
- Another example: helped debug error with my plot.

**Internet Resources:** Stack Exchange, Stack Overflow, etc.

(describe any information that you learned from internet resources, including the URLs)
- data frames in Pandas versus R from StackOverflow  https://stackoverflow.com/questions/8991709/why-were-pandas-merges-in-python-faster-than-data-table-merges-in-r-in-2012

**ChatGPT, Gemini, Claude, etc:** Any language models or generative AI chatbots that helped you.

(if you used any such tools, please tell us here)
- For example:  I asked ChatGPT how to define a new data frames
- Another example:  Gemini told me how to make a function for sorting my data

- ***Link to AI Chat History***: Please share a link to your chat if you used AI (ex. ChatGPT Shared Links). Here is the link: https://chatgpt.com/share/68e4980e-41f0-800d-aa1e-3ddfd0fb5a7c 
**OVERALL MESSAGE:** Any time that you used anything except your brain to solve the questions in these projects, you need to disclose such resources at the start of the project, with details about your usage of the tools.

**YOUR OWN WORK:** Even when you utilize other resources, do NOT just copy and paste.  Write all explanations in your own words, using several sentences in English, which are understandable and which you wrote (and did not just copy and paste).

## Question 1

In [1]:
%sql sqlite:////anvil/projects/tdm/data/lahman/lahman.db

In [2]:
%%sql
SELECT name FROM sqlite_master WHERE type='table';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


name
teamsfranchises
teams
teamshalf
people
allstarfull
appearances
awardsmanagers
awardsplayers
awardssharemanagers
awardsshareplayers


In [3]:
%%sql
SELECT * FROM appearances 
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


yearID,teamID,lgID,playerID,G_all,GS,G_batting,G_defense,G_p,G_c,G_1b,G_2b,G_3b,G_ss,G_lf,G_cf,G_rf,G_of,G_dh,G_ph,G_pr
1871,BS1,NA,barnero01,31,31,31,31,0,0,0,16,0,15,0,0,0,0,0,0,0
1871,BS1,NA,barrofr01,18,17,18,18,0,0,0,1,0,0,13,0,4,17,0,0,0
1871,BS1,NA,birdsda01,29,29,29,29,0,7,0,0,0,0,0,0,27,27,0,0,0
1871,BS1,NA,conefr01,19,18,19,19,0,0,0,0,0,0,18,0,1,19,0,0,0
1871,BS1,NA,gouldch01,31,31,31,31,0,0,30,0,0,0,0,0,1,1,0,0,0


In [4]:
%%sql
SELECT * FROM schools 
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


schoolID,name_full,city,state,country
abilchrist,Abilene Christian University,Abilene,TX,USA
adelphi,Adelphi University,Garden City,NY,USA
adrianmi,Adrian College,Adrian,MI,USA
akron,University of Akron,Akron,OH,USA
alabama,University of Alabama,Tuscaloosa,AL,USA


In [5]:
%%sql
SELECT * FROM parks 
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


parkalias,parkkey,parkname,city,state,country
,ALB01,Riverside Park,Albany,NY,US
,ALT01,Columbia Park,Altoona,PA,US
Edison Field; Anaheim Stadium,ANA01,Angel Stadium of Anaheim,Anaheim,CA,US
,ARL01,Arlington Stadium,Arlington,TX,US
The Ballpark in Arlington; Ameriquest Fl,ARL02,Rangers Ballpark in Arlington,Arlington,TX,US


1.1 The 3 additional tables that I decided to explore are: appearances, schools and parks. 
1.2 There are 3 columns that appear in both the schools and parks tables. These are city, state and country. However, there are many unique columns as well. The appearances table contains columns like yearID, teamID, IgID, player ID, G_all, GS, G_batting, G_defence etc. The schools table contains columns like schoolID, and name_full. The parks table contains columns like parkalias, parkkey and parkname. 
1.3 I feel that the appearances table does not contain anything in common with the schools and parks table. It is completely different and unique. However the schools and parks table might have some association since they have 3 similar columns. For example, each student from the schools table could be living in some city, state and country that matches with the parks table columns of city, state and country. Hence, we can know the parknames closer to the student's locations by identifying them through their parkkey. 

## Question 2

In [6]:
%%sql
SELECT * FROM People
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001
aaronha01,1934,2,5,Mobile,USA,AL,2021,1,22,USA,GA,Atlanta,Hank,Aaron,Henry Louis,180,72,R,R,1954-04-13,aaronha01,1976-10-03,aaroh101
aaronto01,1939,8,5,Mobile,USA,AL,1984,8,16,USA,GA,Atlanta,Tommie,Aaron,Tommie Lee,190,75,R,R,1962-04-10,aaronto01,1971-09-26,aarot101
aasedo01,1954,9,8,Orange,USA,CA,,,,,,,Don,Aase,Donald William,190,75,R,R,1977-07-26,aasedo01,1990-10-03,aased001
abadan01,1972,8,25,Palm Beach,USA,FL,,,,,,,Andy,Abad,Fausto Andres,184,73,L,L,2001-09-10,abadan01,2006-04-13,abada001


In [7]:
%%sql
SELECT * FROM Salaries
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


yearID,teamID,lgID,playerID,salary
1985,ATL,NL,barkele01,870000.0
1985,ATL,NL,bedrost01,550000.0
1985,ATL,NL,benedbr01,545000.0
1985,ATL,NL,campri01,633333.0
1985,ATL,NL,ceronri01,625000.0


In [10]:
%%sql
SELECT * FROM Salaries JOIN People ON Salaries.playerID = People.playerID LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


yearID,teamID,lgID,playerID,salary,playerID_1,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID
1985,ATL,NL,barkele01,870000.0,barkele01,1955,7,7,Fort Knox,USA,KY,,,,,,,Len,Barker,Leonard Harold,225,77,R,R,1976-09-14,barkele01,1987-09-26,barkl001
1985,ATL,NL,bedrost01,550000.0,bedrost01,1957,12,6,Methuen,USA,MA,,,,,,,Steve,Bedrosian,Stephen Wayne,200,75,R,R,1981-08-14,bedrost01,1995-08-09,bedrs001
1985,ATL,NL,benedbr01,545000.0,benedbr01,1955,8,18,Birmingham,USA,AL,,,,,,,Bruce,Benedict,Bruce Edwin,175,73,R,R,1978-08-18,benedbr01,1989-09-11,beneb001
1985,ATL,NL,campri01,633333.0,campri01,1953,6,10,Trion,USA,GA,2013,4,25,USA,GA,Rydal,Rick,Camp,Rick Lamar,195,73,R,R,1976-09-15,campri01,1985-10-05,campr001
1985,ATL,NL,ceronri01,625000.0,ceronri01,1954,5,19,Newark,USA,NJ,,,,,,,Rick,Cerone,Richard Aldo,192,71,R,R,1975-08-17,ceronri01,1992-07-10,ceror001


In [11]:
%%sql
WITH joined_data AS (
    SELECT * FROM Salaries JOIN People ON People.playerID = Salaries.playerID
)

SELECT *, AVG(salary) AS avg_salary FROM joined_data GROUP BY weight ORDER BY weight DESC LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


yearID,teamID,lgID,playerID,salary,playerID:1,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID,avg_salary
2015,CIN,NL,diazju03,510000.0,diazju03,1984,2,27,La Romana,D.R.,La Romana,,,,,,,Jumbo,Diaz,Jose Rafael,315,76,R,R,2014-06-20,diazju03,2017-07-16,diazj005,517500.0
2002,CLE,AL,sabatcc01,700000.0,sabatcc01,1980,7,21,Vallejo,USA,CA,,,,,,,CC,Sabathia,Carsten Charles,300,78,L,L,2001-04-08,sabatc.01,2019-09-24,sabac001,14576190.4
1996,SLN,NL,youngdm01,109000.0,youngdm01,1973,10,11,Vicksburg,USA,MS,,,,,,,Dmitri,Young,Dmitri Dell,295,74,B,R,1996-08-29,youngdm01,2008-07-11,yound001,3771714.285714286
2002,CHA,AL,rauchjo01,200000.0,rauchjo01,1978,9,27,Louisville,USA,KY,,,,,,,Jon,Rauch,Jon Erich,290,83,R,R,2002-04-02,rauchjo01,2013-05-17,raucj001,1613250.0
1997,CLE,AL,colonba01,150000.0,colonba01,1973,5,24,Altamira,D.R.,Puerto Plata,,,,,,,Bartolo,Colon,Bartolo,285,71,R,R,1997-04-04,colonba01,2018-09-22,colob001,5617744.680851064
2002,BOS,AL,pickeca01,200000.0,pickeca01,1976,9,29,St. Thomas,V.I.,St. Thomas,,,,,,,Calvin,Pickering,Calvin Elroy,283,77,L,L,1998-09-12,pickeca01,2005-04-21,pickc001,261750.0
2002,PHI,NL,silvaca01,200000.0,silvaca01,1979,4,23,Bolivar,Venezuela,Bolivar,,,,,,,Carlos,Silva,Carlos,280,76,R,R,2002-04-01,silvaca01,2010-09-07,silvc001,2301871.0
2003,CHN,NL,zambrca01,340000.0,zambrca01,1981,6,1,Puerto Cabello,Venezuela,Carabobo,,,,,,,Carlos,Zambrano,Carlos Alberto,275,76,B,R,2001-08-20,zambrca01,2012-09-21,zambc001,10164293.103448275
2016,MIN,AL,sanomi01,547500.0,sanomi01,1993,5,11,San Pedro de Macoris,D.R.,San Pedro de Macoris,,,,,,,Miguel,Sano,Miguel Angel,272,76,R,R,2015-07-02,sanomi01,2022-07-29,sanom001,547500.0
1999,CHA,AL,leeca01,200000.0,leeca01,1976,6,20,Aguadulce,Panama,Cocle,,,,,,,Carlos,Lee,Carlos,270,74,R,R,1999-05-07,leeca01,2012-10-03,lee-c001,4128199.1228070175


Markdown notes and full English sentences and analysis written here.

## Question 3

In [12]:
%%sql
SELECT * FROM batting 
LIMIT 5

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,yearID,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP
aardsda01,2004,1,SFN,NL,11,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
aardsda01,2006,1,CHN,NL,45,,2,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0
aardsda01,2007,1,CHA,AL,25,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
aardsda01,2008,1,BOS,AL,47,,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0
aardsda01,2009,1,SEA,AL,73,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


In [13]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)
SELECT * FROM joined_data LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID,playerID:1,yearID,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2004,1,SFN,NL,11,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2006,1,CHN,NL,45,,2,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2007,1,CHA,AL,25,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2008,1,BOS,AL,47,,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2009,1,SEA,AL,73,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


In [32]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT SUM(R) AS homeplate_count FROM joined_data WHERE nameFirst = 'Babe' AND nameLast = 'Ruth';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


homeplate_count
2174


In [24]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT *, AVG(R) AS avg_points_scored FROM joined_data WHERE birthYear >= 1980 AND birthYear < 1990 GROUP BY birthYear ORDER BY birthYear;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID,playerID:1,yearID,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP,avg_points_scored
abercre01,1980,7,15,Columbus,USA,GA,,,,,,,Reggie,Abercrombie,Reginald Damascus,215,75,R,R,2006-04-04,abercre01,2008-09-28,aberr001,abercre01,2006,1,FLO,NL,111,,255,39,54,12,2,5,24,6,5,18,78,2,3,4,1,2,17.10182767624021
aardsda01,1981,12,27,Denver,USA,CO,,,,,,,David,Aardsma,David Allan,215,75,R,R,2004-04-06,aardsda01,2015-08-23,aardd001,aardsda01,2004,1,SFN,NL,11,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,14.335890878090366
aceveal01,1982,12,8,San Luis Rio Colorado,Mexico,Sonora,,,,,,,Alfredo,Aceves,Alfredo,205,74,R,R,2008-08-31,aceveal01,2014-06-02,aceva001,aceveal01,2008,1,NYA,AL,6,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,16.083526682134572
alberma01,1983,1,20,Houston,USA,TX,,,,,,,Matt,Albers,Matthew James,225,73,L,R,2006-07-25,alberma01,2019-09-28,albem001,alberma01,2006,1,HOU,NL,4,,4,0,0,0,0,0,0,0,0,0,3,0,0,0,0,0,16.491117478510027
abreuto01,1984,11,13,Puerto Plata,D.R.,Puerto Plata,,,,,,,Tony,Abreu,Etanislao Toni,200,70,B,R,2007-05-22,abreuto01,2014-07-28,abret001,abreuto01,2007,1,LAN,NL,59,,166,19,45,14,1,2,17,0,0,7,21,1,3,0,2,5,13.758029978586723
abadfe01,1985,12,17,La Romana,D.R.,La Romana,,,,,,,Fernando,Abad,Fernando Antonio,235,74,L,L,2010-07-28,abadfe01,2021-10-01,abadf001,abadfe01,2010,1,HOU,NL,22,,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,12.153616703952274
adamsau01,1986,8,19,Montgomery,USA,AL,,,,,,,Austin,Adams,Austin David,205,71,R,R,2014-07-12,adamsau01,2019-07-06,adama001,adamsau01,2014,1,CLE,AL,6,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,13.602470930232558
abreujo02,1987,1,29,Cienfuegos,Cuba,Cienfuegos,,,,,,,Jose,Abreu,Jose Dariel,235,75,R,R,2014-03-31,abreujo02,2022-10-04,abrej003,abreujo02,2014,1,CHA,AL,145,,556,80,176,35,2,36,107,3,1,51,131,15,11,0,4,14,14.006301197227474
achteaj01,1988,8,27,Toledo,USA,OH,,,,,,,A. J.,Achter,Adam Joseph,215,77,R,R,2014-09-03,achteaj01,2016-10-01,achta001,achteaj01,2014,1,MIN,AL,7,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,9.934713375796179
adamsla01,1989,11,13,Talihina,USA,OK,,,,,,,Lane,Adams,Lane Weston,190,76,R,R,2014-09-01,adamsla01,2018-09-30,adaml001,adamsla01,2014,1,KCA,AL,6,,3,1,0,0,0,0,0,0,0,0,2,0,0,0,0,0,12.811985018726592


3.1 I got 2174 as my home plate count as well. I used ChatGPT to debug my code and get an idea of how to approach this question. 
3.2 I referred to project 5 questions for guidance with the calculation part as it is very similar.  

## Question 4

In [26]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT *, SUM(R) AS career_runs FROM joined_data GROUP BY playerID ORDER BY career_runs DESC LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID,playerID:1,yearID,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP,career_runs
henderi01,1958,12,25,Chicago,USA,IL,,,,,,,Rickey,Henderson,Rickey Nelson Henley,180,70,R,L,1979-06-24,henderi01,2003-09-19,hendr001,henderi01,1979,1,OAK,AL,89,,351,49,96,13,3,1,26,33,11,34,39,0,2,8,3,4,2295
cobbty01,1886,12,18,Narrows,USA,GA,1961,7,17,USA,GA,Atlanta,Ty,Cobb,Tyrus Raymond,175,73,L,R,1905-08-30,cobbty01,1928-09-11,cobbt101,cobbty01,1905,1,DET,AL,41,,151,19,36,6,0,1,15,2,,10,23,,0,4,,,2247
bondsba01,1964,7,24,Riverside,USA,CA,,,,,,,Barry,Bonds,Barry Lamar,185,73,L,L,1986-05-30,bondsba01,2007-09-26,bondb001,bondsba01,1986,1,PIT,NL,113,,413,72,92,26,3,16,48,36,7,65,102,2,2,2,2,4,2227
ruthba01,1895,2,6,Baltimore,USA,MD,1948,8,16,USA,NY,New York,Babe,Ruth,George Herman,215,74,L,L,1914-07-11,ruthba01,1935-05-30,ruthb101,ruthba01,1914,1,BOS,AL,5,,10,1,2,1,0,0,2,0,,0,4,,0,0,,,2174
aaronha01,1934,2,5,Mobile,USA,AL,2021,1,22,USA,GA,Atlanta,Hank,Aaron,Henry Louis,180,72,R,R,1954-04-13,aaronha01,1976-10-03,aaroh101,aaronha01,1954,1,ML1,NL,122,,468,58,131,27,6,13,69,2,2,28,39,,3,6,4,13,2174
rosepe01,1941,4,14,Cincinnati,USA,OH,,,,,,,Pete,Rose,Peter Edward,192,71,B,R,1963-04-08,rosepe01,1986-08-17,rosep001,rosepe01,1963,1,CIN,NL,157,,623,101,170,25,9,6,41,13,15,55,72,0,5,6,6,8,2165
mayswi01,1931,5,6,Westfield,USA,AL,,,,,,,Willie,Mays,Willie Howard,170,70,R,R,1951-05-25,mayswi01,1973-09-09,maysw101,mayswi01,1951,1,NY1,NL,121,,464,59,127,22,5,20,68,7,4,57,60,,2,1,,11,2062
rodrial01,1975,7,27,New York,USA,NY,,,,,,,Alex,Rodriguez,Alexander Enmanuel,230,75,R,R,1994-07-08,rodrial01,2016-08-12,rodra001,rodrial01,1994,1,SEA,AL,17,,54,4,11,0,0,0,2,3,0,3,20,0,0,1,1,0,2021
ansonca01,1852,4,17,Marshalltown,USA,IA,1922,4,14,USA,IL,Chicago,Cap,Anson,Adrian Constantine,227,72,R,R,1871-05-06,ansonca01,1897-10-03,ansoc101,ansonca01,1871,1,RC1,NA,25,,120,29,39,11,3,0,16,6,2,2,1,,,,,0,1999
musiast01,1920,11,21,Donora,USA,PA,2013,1,19,USA,MO,Ladue,Stan,Musial,Stanley Frank,175,72,L,L,1941-09-17,musiast01,1963-09-29,musis101,musiast01,1941,1,SLN,NL,12,,47,8,20,4,0,1,7,1,,2,1,,0,0,,0,1949


In [30]:
%%sql
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)

SELECT *, SUM(H) AS career_hits FROM joined_data GROUP BY playerID ORDER BY career_hits DESC LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,birthYear,birthMonth,birthDay,birthCity,birthCountry,birthState,deathYear,deathMonth,deathDay,deathCountry,deathState,deathCity,nameFirst,nameLast,nameGiven,weight,height,bats,throws,debut,bbrefID,finalGame,retroID,playerID:1,yearID,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP,career_hits
rosepe01,1941,4,14,Cincinnati,USA,OH,,,,,,,Pete,Rose,Peter Edward,192,71,B,R,1963-04-08,rosepe01,1986-08-17,rosep001,rosepe01,1963,1,CIN,NL,157,,623,101,170,25,9,6,41,13,15,55,72,0,5,6,6,8,4256
cobbty01,1886,12,18,Narrows,USA,GA,1961,7,17,USA,GA,Atlanta,Ty,Cobb,Tyrus Raymond,175,73,L,R,1905-08-30,cobbty01,1928-09-11,cobbt101,cobbty01,1905,1,DET,AL,41,,151,19,36,6,0,1,15,2,,10,23,,0,4,,,4189
aaronha01,1934,2,5,Mobile,USA,AL,2021,1,22,USA,GA,Atlanta,Hank,Aaron,Henry Louis,180,72,R,R,1954-04-13,aaronha01,1976-10-03,aaroh101,aaronha01,1954,1,ML1,NL,122,,468,58,131,27,6,13,69,2,2,28,39,,3,6,4,13,3771
musiast01,1920,11,21,Donora,USA,PA,2013,1,19,USA,MO,Ladue,Stan,Musial,Stanley Frank,175,72,L,L,1941-09-17,musiast01,1963-09-29,musis101,musiast01,1941,1,SLN,NL,12,,47,8,20,4,0,1,7,1,,2,1,,0,0,,0,3630
speaktr01,1888,4,4,Hubbard,USA,TX,1958,12,8,USA,TX,Lake Whitney,Tris,Speaker,Tristram Edgar,193,71,L,L,1907-09-12,speaktr01,1928-08-30,speat101,speaktr01,1907,1,BOS,AL,7,,19,0,3,0,0,0,1,0,,1,4,,0,0,,,3514
jeterde01,1974,6,26,Pequannock,USA,NJ,,,,,,,Derek,Jeter,Derek Sanderson,195,75,R,R,1995-05-29,jeterde01,2014-09-28,jeted001,jeterde01,1995,1,NYA,AL,15,,48,5,12,4,1,0,7,0,0,3,11,0,0,0,0,0,3465
ansonca01,1852,4,17,Marshalltown,USA,IA,1922,4,14,USA,IL,Chicago,Cap,Anson,Adrian Constantine,227,72,R,R,1871-05-06,ansonca01,1897-10-03,ansoc101,ansonca01,1871,1,RC1,NA,25,,120,29,39,11,3,0,16,6,2,2,1,,,,,0,3435
wagneho01,1874,2,24,Chartiers,USA,PA,1955,12,6,USA,PA,Carnegie,Honus,Wagner,John Peter,200,71,R,R,1897-07-19,wagneho01,1917-09-17,wagnh101,wagneho01,1897,1,LS3,NL,62,,242,38,81,18,4,2,39,20,,15,14,,1,5,,,3420
yastrca01,1939,8,22,Southampton,USA,NY,,,,,,,Carl,Yastrzemski,Carl Michael,175,71,L,R,1961-04-11,yastrca01,1983-10-02,yastc101,yastrca01,1961,1,BOS,AL,148,,583,71,155,31,6,11,80,6,5,50,96,3,3,2,5,19,3419
pujolal01,1980,1,16,Santo Domingo,D.R.,Distrito Nacional,,,,,,,Albert,Pujols,Jose Alberto,235,75,R,R,2001-04-02,pujolal01,2022-10-04,pujoa001,pujolal01,2001,1,SLN,NL,161,,590,112,194,47,4,37,130,1,3,69,93,6,9,1,7,21,3384


4.1 I constructed my query by referring to question 3 of this project. 
4.2 I constructed my query by referring to 4.1 of the question and replacing 'R' with 'H'.  
4.3 There are 5 players that appear in both lists. They are Pete Rose, Ty Cobb, Hank Aaron, Stan Musial and Cap Anson.  

## Question 5

In [33]:
%%sql
SELECT * FROM CollegePlaying LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,schoolID,yearID
aardsda01,pennst,2001
aardsda01,rice,2002
aardsda01,rice,2003
abadan01,gamiddl,1992
abadan01,gamiddl,1993


In [52]:
%%sql
WITH joined_data AS (
    SELECT CollegePlaying.playerID, Batting.R FROM CollegePlaying JOIN Batting ON CollegePlaying.playerID = Batting.playerID  WHERE CollegePlaying.schoolID = 'purdue'
)

SELECT *, SUM(R) AS career_runs FROM joined_data GROUP BY playerID ORDER BY career_runs DESC LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,R,career_runs
allenbe01,2,1071
skowrbi01,0,681
allenje01,9,465
mackife01,0,255
rossyri01,0,172
cianfar01,3,136
tragewa01,0,108
mccabjo01,1,6
kellysp01,3,3
wisehu01,0,0


In [49]:
%%sql
WITH joined_data AS (
    SELECT * FROM CollegePlaying JOIN Batting ON CollegePlaying.playerID = Batting.playerID
)

SELECT *, AVG(R) AS avg_career_runs FROM joined_data GROUP BY schoolID ORDER BY avg_career_runs DESC LIMIT 10; 

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


playerID,schoolID,yearID,playerID:1,yearID:1,stint,teamID,lgID,G,G_batting,AB,R,H,2B,3B,HR,RBI,SB,CS,BB,SO,IBB,HBP,SH,SF,GIDP,avg_career_runs
bagweje01,hartford,1987,bagweje01,1991,1,HOU,NL,156,,554,79,163,26,4,15,82,7,4,75,116,5,13,1,7,12,86.24528301886792
moronje01,txwesleyan,1980,moronje01,1984,1,CLE,AL,7,,19,1,3,1,0,0,3,0,0,3,5,0,0,1,0,0,78.45833333333333
pujolal01,momaple,1999,pujolal01,2001,1,SLN,NL,161,,590,112,194,47,4,37,130,1,3,69,93,6,9,1,7,21,75.1923076923077
wanerpa01,ecentralok,1922,wanerpa01,1926,1,PIT,NL,144,,536,101,180,35,22,8,79,11,,66,19,,4,12,,,73.95454545454545
greerru01,montevallo,1988,greerru01,1994,1,TEX,AL,80,,277,36,87,16,1,10,46,0,0,46,46,2,2,2,4,3,71.44444444444444
grantge01,narizona,1915,grantge01,1922,1,CHN,NL,7,,23,3,4,1,1,0,3,2,0,1,3,,0,0,,,70.15384615384616
beschbo01,wittnbrgoh,1904,beschbo01,1908,1,CIN,NL,32,,114,16,31,5,5,0,17,10,,9,22,,2,2,,,68.0909090909091
grandcu01,illchicago,2000,grandcu01,2004,1,DET,AL,9,,25,2,6,1,1,0,0,0,0,3,8,0,0,0,0,1,67.61111111111111
careyma01,inconco,1907,careyma01,1910,1,PIT,NL,2,,6,2,3,0,1,0,2,0,,2,1,,0,0,,,66.32352941176471
phillto02,nmmilin,1978,phillto02,1982,1,OAK,AL,40,,81,11,17,2,2,0,8,2,3,12,26,0,2,5,0,0,65.0


5.1 I constructed my query by referring to question 4 of this project. 
5.2 The highest number of runs scored by a player from Purdue University is 1071. Their playerID is 'allenbe01'.
5.3 I had some difficulty formulating my query for this part and I used ChatGPT to debug my code, but it did not help. Hence, I constructed my own join using the join formula given in question 2. I then calculated my average career runs similar to how I had done it in the previous few questions. I used the same syntax rules. Even though not specified, I referred to Piazza discussion board and found that I can limit my output to the top 10 rows, showing the highest number of average career runs by university.  
5.4 Based on the output found in 5.3, hartford university has the highest average runs scored by its players. 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.

https://www.purdue.edu/odos/osrr/honor-pledge/
